components/SpexList/SpexList.js

```diff
@@ -1,18 +1,68 @@
+
 import { createClient } from "@/utils/supabase/server";
 import styles from "@/app/page.module.css";
 import Link from "next/link";
+
+const [sorting, setSorting] = ["year", null];
+function sort(spexWithYear) {
+  console.log("start", spexWithYear);
+  spexWithYear.sort((b, a) => {
+    const [yearA, termA] = a.year.split('_');
+    const [yearB, termB] = b.year.split('_');
+  
+    // Jämför år
+    const yearComparison = parseInt(yearA) - parseInt(yearB);
+    if (yearComparison !== 0) {
+      return yearComparison;
+    }
+  
+    // Om åren är lika, jämför terminer
+    if (termA && termB) {
+      return termB.localeCompare(termA);
+    }
+  
+    return 0;
+  });
+  console.log("end", spexWithYear);
+  return spexWithYear;
+}
 async function fetchData() {
   const supabase = createClient();
+  let spex = await supabase.from("spex").select("id, name").order("id");
 
-  return supabase.from("spex").select("id, name").order("id");
-}
+  // Använd Promise.all för att vänta på alla asynkrona operationer
+  const spexWithYear = await Promise.all(
```

Comment:
Denna är för ineffektiv. Kan du inte modifiera queriet ovan för att hämta ut denna datan direkt? Nu körs ju ett query per spex för att hämta ut årtalet.

Om 100 pers på en sittning går in på spärmen körs 4200 queries som hämtar ut samma sak istället för 100.

Reply:
Försökte få det till en query. Men supabase verkar ha en del begräningar med att hämta data från olika tabeller. Men glömde bort att vi hade begränsat med requests iom gratisverion. Ska kolla på det!

Reply:
Tycker du löste det snyggt i sökmetoden, kanske nåt liknande funkar här?